nullaway/src/main/java/com/uber/nullaway/handlers/FieldInitializationSerializationHandler.java

```diff
@@ -47,23 +59,27 @@
     this.config = config;
   }
 
-  /**
-   * If the method guarantees to leave the initialized class field to be {@code @Nonnull} at exit
-   * point, this method will serialize information regarding the initializer method and the class
-   * field. Since traversing AST is costly, we do it only inside the handler when the feature is
-   * enabled and, this method accesses the initializer method through the leaf node in state
-   * parameter.
-   */
   @Override
   public void onNonNullFieldAssignment(
       Symbol field, AccessPathNullnessAnalysis analysis, VisitorState state) {
+    // Traversing AST is costly, therefore we access the initializer method through the leaf node in
+    // state parameter
     TreePath pathToMethod =
         NullabilityUtil.findEnclosingMethodOrLambdaOrInitializer(state.getPath());
     if (pathToMethod == null || pathToMethod.getLeaf().getKind() != Tree.Kind.METHOD) {
       return;
     }
     Symbol.MethodSymbol symbol = (Symbol.MethodSymbol) ASTHelpers.getSymbol(pathToMethod.getLeaf());
-    // We are only looking for non-constructor methods that initializes a class field.
+
+    // Check if the method and the field are in the same class.
+    ClassTree methodEnclosingClass = ASTHelpers.findEnclosingNode(pathToMethod, ClassTree.class);
```

Comment:
You already have symbols here. It's faster to use the symbol hierarchy than traversing the AST, I think. Take a look at `Symbol.enclClass()`. It should also be faster to compare two `ClassSymbol` objects than two `ClassTree` objects.

Reply:
b02bb563c90b4caf22c86bc15a58f252bd8d6dc2